### ISSUE TITLE
AutoRestart Excavator on freeze.

### ISSUE THREAD

**Issue by SystemtechCA** [cTn-dev/Excavator-GUI#10]
So on the odd occasion i have had excavator freeze or throw a error of some sort. Now thats excavator problem, not your obviously, but seeing as you can see the hash rate for the worker etc, it would be very very cool that if a job is running and the hash rate = 0 (or maybe less that 10%) then stop excavator and restart it and the job.

Thoughts?
Thanks for the AWESOME GUI!!!!

**Comment by cTn-dev** [comment]
Right now there is an auto-restart functionality in the GUI, if excavator encounters a critical error, the GUI will start a new instance of excavator and restart all the jobs.

I thought that the current implementation is more or less bullet proof when it comes to such behavior, if excavator crashes it gets restarted, if pool goes down, backup pool will take over.

Have you encountered a different type of "crash" which left excavator in a broken state (not mining, but not restarting either) ?

**Comment by SystemtechCA** [comment]
Yes, i have 8GB RAM on my mining PC which should be enough but occasionally excavator gets a out of memory or a memory related error and therefore my mining gets stuck for a few hours until i notice. The pool stays connected but the hash rate =0 so the nicehash notifications don't get sent either as its technically still connected.

**Comment by cTn-dev** [comment]
Do you by any chance have a log for when this thing occurred? It might be better to ask the excavator devs about this first, as i am guessing if the mining dies but the pools are still connected, there is a decent chance the API will be either unresponsive or unreliable.
The excavator devs could consider an critical bug since their auto-restart script provided with excavator is based on the http API being alive, if crash like this occurs, the auto-restart script would never trigger.

With current GUI implementation it is rather difficult to kill the the excavator process while the API is down.

If the excavator devs would dismiss issue like this i would look into it on the GUI end, but it would be difficult to implement reliably (as of now).

I am guessing you already tried simple solutions like increasing the virtual paging file in windows to make excavator less prone to go over the bounds?

**Comment by SystemtechCA** [comment]
Ok im with you. Makes sense. I will look for logs of sort or get a screenshot of the error the next time it happens as i cannot find any logs anywhere in excavator.
Thank you though.

**Comment by cTn-dev** [comment]
Sure, keep me updated.
Word of warning, since the GUI now supports configuring file logging parameters, be careful with extensive settings, i managed to lock up the excavator API with file logging set at 0 and GUI running.
Excavator will continue mining but any action done in GUI will have no effect, unfortunately so far i have no info about this from the excavator devs :(
